### PR TITLE
[14.0][ADD] shopinvader_multi_cart

### LINKED ISSUE #999: [RFC] Multiple carts per user
**Requirement:** have the possibility to handle multiple carts per user.
**Use case:** I can prepare different carts for different dates at the same time
**Usage:**
* I can go to `/cart/list` and see the list of my carts
* I can select a cart (call `/cart/select`, change session cart ID)
  * shall we call `customer._assign_cart_and_get_store_cache`? 
  * when the order is selected shall we recompute totals etc? (something might change if I play w/ an order after days)
* I can create a new cart (just flush session cart ID then rely on next `add_item` to create a new one)
* I can delete a cart (call `/cart/delete/$id`)

Any comments / recommendations / remarks?

### PR DESCRIPTION
**Pull request by ivantodorovich**
Closes https://github.com/shopinvader/odoo-shopinvader/issues/999

### CODE REVIEW

File: shopinvader_multi_cart/services/carts.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        """Deletes the cart with the given id.
+        :param _id: int
+        """
+        cart = self._get(_id)
+        cart.unlink()
+        if self.shopinvader_session.get("cart_id") == _id:
+            return self._clear_session_cart()
+        return {}
+
+    def new(self):
```
Comment (simahawk) [inline shopinvader_multi_cart/services/carts.py]: is this really useful?

**Review by simahawk — CHANGES_REQUESTED** [review]
LG overall. Tests are red.

**Comment by simahawk** [comment]
for tests, rebase once #1016 is merged. Should fix them.

File: shopinvader_multi_cart/services/carts.py (hunk 1 above)
Comment (ivantodorovich) [reply shopinvader_multi_cart/services/carts.py]: Hmm, I think so.
I'm trying to cover this from the specs:

> I can create a new cart (just flush session cart ID then rely on next add_item to create a new one)

How would the user create a second cart without this? It's the only way to have len(carts) > 1 IMO

**Review by sebastienbeau — APPROVED** [review]
LGTM (code review).
thanks for your work

File: shopinvader_multi_cart/services/carts.py (hunk 1 above)
Comment (simahawk) [reply shopinvader_multi_cart/services/carts.py]: yep, but as you see I didn't state an endpoint for this. What I meant, was: in the frontend I can get a new cart by simply dropping the cart ID in the session and by doing so I'll get a new cart automatically when calling `/add_item`.
Sounds good?

Comment (sebastienbeau) [reply shopinvader_multi_cart/services/carts.py]: Indeed. In frontend we have access to the session id of the cart so we can remove it easily

Comment (sebastienbeau) [reply shopinvader_multi_cart/services/carts.py]: so let's remove it

Comment (ivantodorovich) [reply shopinvader_multi_cart/services/carts.py]: Understood :+1: 
I've just removed it :)

**Review by simahawk — APPROVED** [review]

**Pull request merged**